fsharp-backend/src/BackendOnlyStdLib/LibHttpBytesClient.fs

```diff
@@ -0,0 +1,82 @@
+/// StdLib functions in the HttpBytesClient module
+module BackendOnlyStdLib.LibHttpBytesClient
+
+open System.Net.Http
+
+open Prelude
+open LibExecution
+open LibExecution.RuntimeTypes
+
+module Errors = LibExecution.Errors
+
+let incorrectArgs = Errors.incorrectArgs
+
+let fn = FQFnName.stdlibFnName
+
+// Converts an object to (string, string) pairs
+let toStringPairs (dv : Dval) : Result<List<string * string>, string> =
+  match dv with
+  | DList tpls ->
+    tpls
+    |> List.map (fun pair ->
+      match pair with
+      | DTuple (DStr k, DStr v, []) ->
+        // TODO trim key and/or value here?
+        // TODO error on empty key?
+        Ok(k, v)
+      | other ->
+        Error
+          $"Expected a (string * string), but got: {DvalReprDeveloper.toRepr other}")
+    |> Tablecloth.Result.values
+  | _ -> Error $"Expected a list of tuples, but got: {DvalReprDeveloper.toRepr dv}"
+
+let call (method : HttpMethod) =
+  (function
+  | _, [ DStr uri; DBytes body; query; headers ] ->
+    let headers = toStringPairs headers
+    let query = HttpQueryEncoding.toQuery query
```

Comment:
Perhaps the query should be a string?

Reply:
perhaps it shouldn't exist, and the query should be included by the user in the URL. That's what Elm does, it seems: https://package.elm-lang.org/packages/elm/http/latest/Http

Reply:
Yes, that makes sense to me.